### ISSUE TITLE
Error while accessing nextcloud through document provider in another app when device is offline

### ISSUE THREAD

**Issue by adityamj** [nextcloud/ios#204]
Device: Ipad 2017 model
IOS version: 10.3.1
App version: 2.17.1.000001

Issue: "The Internet connection appears to be offline." when device is not connected to the internet and nextcloud is accessed from another application via document provider. 
This happens even when some files are made offline by marking as favourite. 
The files are visible and accessible through nextcloud app however.

**Comment by marinofaggiana** [comment]
Yes, this is correct, document provider need the connection with Nextcloud server for see all directory/files available.

**Comment by adityamj** [comment]
So is this the intended behaviour or a bug, since the cache is already present in the nextcloud app which also shows the directory listing. 
Owncloud on the other hand allows using cached directory listing for offline access. 

On the other hand, is it possible to just make Favourites available through document provider when device is offline?

**Comment by janprzy** [comment]
I'm also experiencing this, it is very annoying. The directory structure should always be cached locally so only the files have to be loaded when the user wants to access them.
iCloud Drive, for example, always lets me navigate my folders. It just shows an icon next to files that need to be downloaded if I want to access them.
